projects/sandbox/train/train/data_structures.py

```diff
@@ -76,27 +131,37 @@ def __next__(self) -> Tuple[torch.Tensor, torch.Tensor]:
 
 
 class BBHNetWaveformInjection(RandomWaveformInjection):
-    def __init__(self, *args, **kwargs):
-        try:
-            glitch_prob = kwargs.pop("glitch_prob")
-            downweight = kwargs.pop("downweight")
-            prob = kwargs.pop("prob")
-        except KeyError:
-            self.downweight = 1
-        else:
-            # not to sound like Fermat but I have a
-            # derivation of this somewhere that I
-            # can't quite find at the moment
-            prob = prob / (1 - glitch_prob * (1 - downweight)) ** 2
-            if not 0 < prob <= 1.0:
-                raise ValueError(
-                    "Probability must be between 0 and 1. Adjust the value(s) "
-                    "of waveform_prob, glitch_prob, and/or downweight"
-                )
-            kwargs["prob"] = prob
-            self.downweight = downweight
+    def __init__(
+        self,
+        *args,
+        prob: float = 1.0,
+        swap_frac: float = 0.0,
+        mute_frac: float = 0.0,
+        downweight: float = 1.0,
+        glitch_prob: float = 0.5,
+        **kwargs
+    ):
+        self.downweight = downweight
+        self.prob = prob
+        # not to sound like Fermat but I have a
+        # derivation of this somewhere that I
+        # can't quite find at the moment
+        prob = prob / (1 - glitch_prob * (1 - downweight)) ** 2
+        # account for the fact that some waveforms will have a channel
+        # swapped with another waveform and labeled as noise
+        prob = prob / (1 - (swap_frac + mute_frac))
```

Comment:
I think this should be `prob = prob / (1 - (swap_frac + mute_frac - swap_frac * mute_frac))` because both can happen

Reply:
agreed